src/data/mystery-encounters/encounters/lost-at-sea-encounter.ts

```diff
@@ -0,0 +1,146 @@
+  .withQuery(`${namepsace}:query`)
+  .withOption(
+    // Option 1: Use a (non fainted) pokemon that can learn Surf to guide you back/
+    new MysteryEncounterOptionBuilder()
+      .withPokemonCanLearnMoveRequirement(OPTION_1_REQUIRED_MOVE)
+      .withOptionMode(EncounterOptionMode.DISABLED_OR_DEFAULT)
+      .withDialogue({
+        buttonLabel: `${namepsace}:option:1:label`,
+        disabledButtonLabel: `${namepsace}:option:1:label_disabled`,
+        buttonTooltip: `${namepsace}:option:1:tooltip`,
+        disabledButtonTooltip: `${namepsace}:option:1:tooltip_disabled`,
+        selected: [
+          {
+            text: `${namepsace}:option:1:selected`,
+          },
+        ],
+      })
+      .withOptionPhase(async (scene: BattleScene) => handlePokemongGuidingYouPhase(scene))
+      .build()
+  )
+  .withOption(
+    //Option 2: Use a (non fainted) pokemon that can learn fly to guide you back.
+    new MysteryEncounterOptionBuilder()
+      .withPokemonCanLearnMoveRequirement(OPTION_2_REQUIRED_MOVE)
+      .withOptionMode(EncounterOptionMode.DISABLED_OR_DEFAULT)
+      .withDialogue({
+        buttonLabel: `${namepsace}:option:2:label`,
+        disabledButtonLabel: `${namepsace}:option:2:label_disabled`,
+        buttonTooltip: `${namepsace}:option:2:tooltip`,
+        disabledButtonTooltip: `${namepsace}:option:2:tooltip_disabled`,
+        selected: [
+          {
+            text: `${namepsace}:option:2:selected`,
+          },
+        ],
+      })
+      .withOptionPhase(async (scene: BattleScene) => handlePokemongGuidingYouPhase(scene))
+      .build()
+  )
+  .withSimpleOption(
+    // Option 3: Wander aimlessly
+    {
+      buttonLabel: `${namepsace}:option:3:label`,
+      buttonTooltip: `${namepsace}:option:3:tooltip`,
+      selected: [
+        {
+          text: `${namepsace}:option:3:selected`,
+        },
+      ],
+    },
+    async (scene: BattleScene) => {
+      const { mysteryEncounter } = scene.currentBattle;
+      const allowedPokemon = scene.getParty().filter((p) => p.isAllowedInBattle());
+
+      for (const pkm of allowedPokemon) {
+        const percentage = DAMAGE_PERCENTAGE / 100;
+        const damage = Math.floor(pkm.getMaxHp() * percentage);
+        applyDamageToPokemon(pkm, damage);
+
+        if (pkm.isFainted()) {
+          mysteryEncounter.setDialogueToken("pokemonNameWithAffix", getPokemonNameWithAffix(pkm));
+          await showEncounterText(scene, "battle:fainted");
+        }
+      }
+
+      leaveEncounterWithoutBattle(scene);
+
+      return true;
+    }
+  )
+  .withOutroDialogue([
+    {
+      text: `${namepsace}:outro`,
+    },
+  ])
+  .build();
+
+/**
+ * Generic handler for using a guiding pokemon to guide you back.
+ *
+ * @param scene Battle scene
+ * @param guidePokemon pokemon choosen as a guide
+ */
+function handlePokemongGuidingYouPhase(scene: BattleScene) {
+  /** Base EXP value for guiding pokemon. Currently Lapras base-value */
+  const baseExpValue: number = 187;
```

Comment:
not that this value should ever change, it might be better to use:
`getPokemonSpecies(Species.LAPRAS).baseExp`

Idk if exp balances are ever adjusted, and to be slightly more readable.

Reply:
OH neat. I didn't know this was available. way better than this thin-air value